src/anyvar/storage/postgres.py

```diff
@@ -85,67 +86,44 @@ def add_objects(self, objects: Iterable[types.VrsObject]) -> None:
         :raise IncompleteVrsObjectError: if object is missing required properties or if
             required properties aren't fully dereferenced
         """
-        objects_list = list(objects)
+        objects_list: list[types.VrsObject] = list(objects)
         if not objects_list:
             return
 
         # Collect unique entities by ID to avoid duplicates
-        sequence_references = {}
-        locations = {}
-        alleles = {}
+        vrs_objects = defaultdict(dict)
 
         # Process all objects and extract their components
         for vrs_object in objects_list:
             try:
-                db_entity = mapper_registry.to_db_entity(vrs_object)
+                db_entity: orm.Base = mapper_registry.to_db_entity(vrs_object)
             except AttributeError as e:
                 raise IncompleteVrsObjectError from e
 
-            if isinstance(db_entity, orm.Allele):
-                alleles[db_entity.id] = db_entity
-                # Also collect the nested location and sequence reference
-                if db_entity.location:
-                    locations[db_entity.location.id] = db_entity.location
-                    if db_entity.location.sequence_reference:
-                        sequence_references[
-                            db_entity.location.sequence_reference.id
-                        ] = db_entity.location.sequence_reference
-            elif isinstance(db_entity, orm.Location):
-                locations[db_entity.id] = db_entity
-                if db_entity.sequence_reference:
-                    sequence_references[db_entity.sequence_reference.id] = (
-                        db_entity.sequence_reference
-                    )
-            elif isinstance(db_entity, orm.SequenceReference):
-                sequence_references[db_entity.id] = db_entity
-            else:
-                raise ValueError(f"Unsupported object type: {type(db_entity)}")  # noqa: TRY004
+            object_parts = db_entity.disassemble()
+            for entity_type, entity in object_parts.items():
+                vrs_objects[entity_type][entity.id] = entity
+
+        # Sort objects so that we insert in dependency order: sequence_references -> locations -> alleles
+        insert_order: list[str] = [
+            orm.SequenceReference.__name__,
+            orm.Location.__name__,
+            orm.Allele.__name__,
+        ]
+        sorted_vrs_objects: dict[str, dict[str, orm.Base]] = {
+            key: vrs_objects[key] for key in insert_order if key in vrs_objects
+        }
```

Comment:
Any reason why we can't do this the previous way (with separate variables) or in a dict with predefined fields?

Reply:
Oh, also, it seems like in some cases sqlalchemy can handle the dependency order itself within a session if we are declaring relationships properly? do we know that we need to be doing this sorting stuff on our side?